teps/0077-pipeline-partial-execution.md

```diff
@@ -0,0 +1,496 @@
+of the file, but general guidance is to include at least TEP number in the
+file name, for example, "/teps/images/NNNN-workflow.jpg".
+-->
+
+## Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+## Design Evaluation
+<!--
+How does this proposal affect the reusability, simplicity, flexibility 
+and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
+-->
+
+## Drawbacks
+
+<!--
+Why should this TEP _not_ be implemented?
+-->
+
+## Future Work
+
+1. **Provide an option to disable a task but not all the that depend on it.**
+    
+    ```yaml
+    spec:
+      pipelineRef:
+        name: a-pipeline
+      disableTasks:
+        - name: B
+          disableDependents: false
+          results:
+            - name: result-A
+              value: result-value
+        - name: G
+    ```
+
+    Above example will run tasks: A, E, F, D, C.
+    Note: E is executed, because `disableDependents: false` and results are provided.
+    
+    Use cases include:
+    1. A user can override the results being referenced from previous Run, by using
+       this section.
+    2. Suppose a failed task is permanently failing, and one would like to disable
+       that task but not its dependents, then the user can hard code the results and 
+       execute the dependents.
+       
+
+2. **Option to include all tasks from previous pipelineRun**
+    ```yaml
+    spec:
+      pipelineRunRef:
+        name: prev-pipeline-run
+        include: All # one of: All|failed|none
+      # One of the failed task is disabled by disableTasks section.
+      disableTasks:
+        - name: B
+        - name: G
+    ```
+    Above example will run tasks: A, F, D, C.
+    By default, only failed and unifinished tasks are included from previous 
+    `PipelineRun`. With this option, a user will be able to include either
+   everything or nothing as well. These are merely convenience options. The
+   same can be achieved by a combination of `disableTasks` and `enableTasks`.
+   
+    Use cases:
+    1. _A convenience option_, if there are large no. of tasks in the previous
+        `PipelineRun` and only a few of them have to be explicitly disabled.
+       Then, one can `include: All` and disable specific ones.
+       
+       Without this option, a user will have to list out all the tasks that
+       should be disabled in the `disableTasks` section.
+
+
+## Alternatives
```

Comment:
There is additional alternatives, and one would be that this is handled by a higher level construct/abstraction, such as [Tekton workflows](https://github.com/tektoncd/community/issues/464) for example.